src/main/resources/public/jsx/DatasetAdder.jsx

```diff
@@ -134,41 +138,62 @@ class DatasetConfigurer extends React.Component {
                 description: '',
                 license: '',
                 source: '',
-                tags: []
-            }
+                tags: [],
+                excludedFields: []
+            },
+            dataset: []
         }
         this.onFieldChange = this.onFieldChange.bind(this)
         this.handleAddition = this.handleAddition.bind(this)
         this.handleDelete = this.handleDelete.bind(this)
-        this.onNameChange = this.onNameChange.bind(this)
+        this.toggleCheckbox = this.toggleCheckbox.bind(this)
+    }
+    componentWillMount(){
+        this.selectedCheckboxes = new Set();
+        this.setState({ dataset : this.props.dataset})
```

Comment:
If you have the dataset in the props, use it from here, don't copy it in the state.